controller/internal/enforcer/nfqdatapath/datapath_udp.go

```diff
@@ -372,8 +375,19 @@ func (d *Datapath) processApplicationUDPSynPacket(udpPacket *packet.Packet, cont
 		return fmt.Errorf("No target found")
 	}
 
+	// We now generate the claims header
+	compactPKI, ok := d.secrets.(*secrets.CompactPKI)
+	if !ok {
+		zap.L().Error("Secrets does not hold compactPKI type, so type assertion failed", zap.Reflect("type", reflect.TypeOf(d.secrets)))
+		return secrets.ErrNotCompactPKI
+	}
+	claimsHeaderBytes := claimsheader.NewClaimsHeader(
+		claimsheader.OptionCompressionType(compactPKI.Compressed),
+		claimsheader.OptionDatapathVersion(d.datapathVersion),
+	).ToBytes()
+
 	udpOptions := d.CreateUDPAuthMarker(packet.UDPSynMask)
```

Comment:
same here. all this code is cut and paste.

Reply:
fixed